var/spack/repos/builtin/packages/xcfun/package.py

```diff
@@ -11,11 +11,14 @@ class Xcfun(CMakePackage):
 
     homepage = "https://github.com/dftlibs/xcfun"
     url = "https://github.com/dftlibs/xcfun/archive/v2.0.0a4.tar.gz"
+    git = "https://github.com/dftlibs/xcfun"
 
     maintainers = ['robertodr', 'bast']
 
+    version('master', branch='master')
```

Comment:
Have you tried the stable 2.0.1 version? If that doesn't work either, could you report (here or on xcfun's repo) what's not compatible/working? I'd rather not have curated packages depend on `master` which is a moving target.

Reply:
Unfortunately, it seems like most recent version of `xcfun` will not compile with `PySCF`, including the recent 2.0.1 release. 

I am OK with removing `master` version from `xcfun`. I do not need it. I only added it because I thought its exclusion was an oversight. If is that the primary objection in the PR, I can certainly remove it.

Reply:
Spack will still use 2.0.0a6 by default. But yes, it would be good to get a stable release that works so we don't have to rely on some random commit (pyscf).

Reply:
@adamjstewart The random commit (pyscf) does indeed look very odd. Based on some perfunctory testing, it looks like it might work with 2.0.0a2. At least it will compile with it, but versions after that fail to compile with PySCF. Unfortunately, I am not a PySCF developer / user, the package contribution was requested by @prckent 

@robertodr Is the random commit (pyscf) also an issue? If I were to remove it, I would have to add version 2.0.0a2 which also looks a bit odd.

Reply:
2.0.0a2 would be better than a random commit in my mind.

Reply:
I would be happy to help restoring compilation either PySCF-side or XCFun-side (if this is what we want to do) but would need some instructions on how to reproduce the problem.

Reply:
Here is how you reproduce the error that I am seeing. First, generalize the dependency on `xcfun` in the py-pyscf Spack package.
```
[naromero@naromero-desktop spack (PySCF_NewPackage)]$ git diff
diff --git a/var/spack/repos/builtin/packages/py-pyscf/package.py b/var/spack/repos/builtin/packages/py-pyscf/package.py
index dc101dc69..f993f6805 100644
--- a/var/spack/repos/builtin/packages/py-pyscf/package.py
+++ b/var/spack/repos/builtin/packages/py-pyscf/package.py
@@ -27,7 +27,7 @@ class PyPyscf(PythonPackage):
     depends_on('blas')
     depends_on('libcint+coulomb_erf+f12')
     depends_on('libxc')
-    depends_on('xcfun@pyscf')
+    depends_on('xcfun')
 
     def setup_build_environment(self, env):
         # Tell PSCF where supporting libraries are located."
```

Then do:
```
[naromero@naromero-desktop spack (PySCF_NewPackage)]$ spack install py-pyscf%gcc@7.5.0 ^python@3.7.7 ^xcfun@2.0.1
```
For any version of `xcfun` besides `2.0.0a1` and `2.0.0a2`.

Here is the error:
```
     4235    /nfs/gce/projects/naromero-workspace/spack/lib/spack/env/gcc/gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall 
             -fPIC -fPIC -I. -Ibuild -I/scratch/naromero/tmp/spack-stage/spack-stage-py-pyscf-1.7.3-a46zaegoegsd75o56mb67tj5ydqhjl7q/spack-src
             /pyscf/lib -I/nfs/gce/software/custom/linux-ubuntu18.04-x86_64/intel/19.0/compilers_and_libraries_2019.0.117/linux/mkl -I/nfs/gce
             /software/custom/linux-ubuntu18.04-x86_64/intel/19.0/compilers_and_libraries_2019.0.117/linux/mkl/include -I/nfs/gce/software/cus
             tom/linux-ubuntu18.04-x86_64/intel/19.0/compilers_and_libraries_2019.0.117/linux/mkl/lib -I/scratch/naromero/opt/spack/linux-ubun
             tu18.04-westmere/gcc-7.5.0/libcint-3.0.13-db5glwr7v4qvctap7o2l22fzavt6ohtw/include -I/scratch/naromero/opt/spack/linux-ubuntu18.0
             4-westmere/gcc-7.5.0/libxc-4.3.2-huxgpctur4ekbyixfuh4ladziyca6qdh/include -I/scratch/naromero/opt/spack/linux-ubuntu18.04-westmer
             e/gcc-7.5.0/xcfun-2.0.1-qdtu7chnlvunxpdhj6i3px5oy2zunpij/include/XCFun -I/scratch/naromero/tmp/spack-stage/spack-stage-py-pyscf-1
             .7.3-a46zaegoegsd75o56mb67tj5ydqhjl7q/spack-src/pyscf/lib/dft -I/scratch/naromero/opt/spack/linux-ubuntu18.04-westmere/gcc-7.5.0/
             python-3.7.7-ageav6epjrvjuwea2r73uwuyocvil2xa/include/python3.7m -c pyscf/lib/dft/xcfun_itrf.c -o build/temp.linux-x86_64-3.7/pys
             cf/lib/dft/xcfun_itrf.o -fopenmp
  >> 4236    pyscf/lib/dft/xcfun_itrf.c:28:20: error: unknown type name 'xc_functional'
     4237     static int eval_xc(xc_functional fun, int deriv, enum xc_vars vars,
     4238                        ^~~~~~~~~~~~~
  >> 4239    pyscf/lib/dft/xcfun_itrf.c:36:22: error: expected end of line before 'default'
     4240     #pragma omp parallel default(none) \
     4241                          ^~~~~~~
  >> 4242    pyscf/lib/dft/xcfun_itrf.c:40:17: error: expected end of line before 'nowait'
     4243     #pragma omp for nowait schedule(static)
     4244                     ^~~~~~
     4245    pyscf/lib/dft/xcfun_itrf.c: In function 'XCFUN_eval_xc':
  >> 4246    pyscf/lib/dft/xcfun_itrf.c:57:9: error: unknown type name 'xc_functional'
     4247             xc_functional fun = xc_new_functional();
     4248             ^~~~~~~~~~~~~
     4249    pyscf/lib/dft/xcfun_itrf.c:57:29: warning: implicit declaration of function 'xc_new_functional' [-Wimplicit-function-declaration]
     4250             xc_functional fun = xc_new_functional();
     4251                                 ^~~~~~~~~~~~~~~~~
     4252    pyscf/lib/dft/xcfun_itrf.c:59:24: warning: implicit declaration of function 'xc_enumerate_parameters'; did you mean 'xcfun_enumer
             ate_parameters'? [-Wimplicit-function-declaration]

     ...

     4274                                xcfun_is_gga
     4275    pyscf/lib/dft/xcfun_itrf.c:154:9: warning: implicit declaration of function 'xc_free_functional'; did you mean '__fortify_functio
             n'? [-Wimplicit-function-declaration]
     4276             xc_free_functional(fun);
     4277             ^~~~~~~~~~~~~~~~~~
     4278             __fortify_function
     4279    pyscf/lib/dft/xcfun_itrf.c: In function 'XCFUN_xc_type':
  >> 4280    pyscf/lib/dft/xcfun_itrf.c:164:9: error: unknown type name 'xc_functional'
     4281             xc_functional fun = xc_new_functional();
     4282             ^~~~~~~~~~~~~
     4283    pyscf/lib/dft/xcfun_itrf.c:165:28: warning: initialization makes pointer from integer without a cast [-Wint-conversion]
     4284             const char *name = xc_enumerate_parameters(fn_id);
     4285                                ^~~~~~~~~~~~~~~~~~~~~~~
  >> 4286    error: command '/nfs/gce/projects/naromero-workspace/spack/lib/spack/env/gcc/gcc' failed with exit status 1

```

I have not dugged into it, not sure if the problem is a header file or that `-fopenmp` is being injected into the compiler line.

Reply:
The problem is that, starting with version 2.0.0a3 the API changed ([migration guide](https://xcfun.readthedocs.io/en/latest/migration.html)) This is a change that PySCF should adapt to, but it's not something to be done within the Spack package itself. I suggest you update the XCFun package to use the 2.0.0a2 tag and open a PySCF issue prompting the developers to migrate to the official version 2 API.

Reply:
ok